src/richie/plugins/lti_consumer/models.py

```diff
@@ -0,0 +1,91 @@
+"""
+LTI consumer plugin models
+"""
+from urllib.parse import unquote
+
+from django.conf import settings
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from cms.models.pluginmodel import CMSPlugin
+from oauthlib import oauth1
+
+from richie.apps.courses.models import Course
+
+
+class LTIConsumer(CMSPlugin):
+    """
+    LTI consumer plugin model.
+    """
+
+    url = models.URLField(_("LTI url"))
+    course = models.ForeignKey(to=Course, on_delete=models.PROTECT)
+    edit_mode = False
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.url:
+            self.url = getattr(settings, "LTI_CONSUMER", {}).get("base_url")
+
+    @property
+    def auth_parameters(self):
+        """
+        Builds required parameters for LTI authentication
+        """
+        role = "instructor" if self.edit_mode else "student"
+        return {
+            "lti_message_type": settings.LTI_CONSUMER.get("display_name"),
```

Comment:
if lti_service is None, we probably need a field on the plugin for this info as well? Ideally, these fields need to disappear when an lti_service is selected from the list of predefined services ;-)